internal/crossdock/client/errorshttpclient/behavior.go

```diff
@@ -150,8 +150,7 @@ func Run(t crossdock.T) {
 			},
 			body:       `{"token":"10"}`,
 			wantStatus: 501,
-			wantBody: `unrecognized procedure ` +
-				`"echo" for service "not-yarpc-test"` + "\n",
+			wantBody:   `unrecognized service name "not-yarpc-test", available services: "yarpc-test"` + "\n",
```

Comment:
can we add a test with multiple services in the list of available services?

Reply:
In order to do this, we need to register a procedure with another service name in dispatcher. The proper place to do this is within `Start()` function of yarpc server. However, the side effect of this method is that all the test cases now have two available service names. I don't know if it's acceptable, appreciated for other work arounds.

Reply:
is there a different place we can add the test then?

Reply:
Also added a test cover this case in router_test.go. If this way is more acceptable, I will remove the change in crossdock test for server setup part.

Reply:
remove crossdock server changes.